stubs/portpicker/portpicker.pyi

```diff
@@ -0,0 +1,21 @@
+import socket
+from typing_extensions import TypeAlias
+
+_Port: TypeAlias = int
+
+__all__ = ("bind", "is_port_free", "pick_unused_port", "return_port", "add_reserved_port", "get_port_from_port_server")
+
+class NoFreePortFoundError(Exception): ...
+
+def add_reserved_port(port: _Port) -> None: ...
+def return_port(port: _Port) -> None: ...
+def bind(port: _Port, socket_type: socket.SocketKind, socket_proto: int) -> _Port | None: ...
+def is_port_free(port: _Port) -> bool: ...
+def pick_unused_port(pid: int | None = None, portserver_address: str | tuple[str, _Port] | None = None) -> _Port: ...
+def get_port_from_port_server(portserver_address: str | tuple[str, _Port], pid: int | None = None) -> _Port | None: ...
```

Comment:
Are you sure it's okay to pass `tuple`s for `portserver_address`? I skimmed through the source code, and it looks to me like it only accepts a `str` for that argument. That seems to be borne out by my REPL experiments, too:

```pycon
Python 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import portpicker
>>> portpicker.pick_unused_port(portserver_address=('foo', 'bar'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 223, in pick_unused_port
    return _pick_unused_port(pid, portserver_address)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 247, in _pick_unused_port
    port = get_port_from_port_server(portserver_address, pid=pid)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 435, in get_port_from_port_server
    buf = _windows_get_port_from_port_server(portserver_address, pid)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 389, in _windows_get_port_from_port_server
    handle = _winapi.CreateFile(
             ^^^^^^^^^^^^^^^^^^^
TypeError: CreateFile() argument 1 must be str, not tuple
>>> portpicker.get_port_from_port_server(portserver_address=('foo', 'bar'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 435, in get_port_from_port_server
    buf = _windows_get_port_from_port_server(portserver_address, pid)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\typeshed\.venv\Lib\site-packages\portpicker.py", line 389, in _windows_get_port_from_port_server
    handle = _winapi.CreateFile(
             ^^^^^^^^^^^^^^^^^^^
TypeError: CreateFile() argument 1 must be str, not tuple
```

Reply:
I believe it accepts a tuple on OSes that are not Windows and that don't support Unix sockets. That's probably a very small set. I can just narrow it to str.